File: codes/DSA 2023/7_duplicate.cpp
// find duplicates

#include <iostream>
using namespace std;

void find_duplicates(int n, int arr[]){

    int count =0; 

    for (int i=0; i<n; i++){
        for(int j=i+1; j<n;j++){
            if(arr[i]==arr[j]){
              cout<<"duplicate found which is "<<arr[j]<<endl;
            }
        }
    }

}
int main(){

    int n;
    cout<<"enter the length of array ";
    cin>>n;
    int arr[n];
    cout<<"enter elements ";
    for(int i=0; i<n; i++){
        cin>>arr[i];
    }
    find_duplicates(n,arr);
     return 0;
}

File: codes/DSA 2023/7_duplicate.py
l = [2,4,3,2,5,3,6]
def find_duplicates(list):
    for i in range(len(list)-1):
        for j in range(i+1,len(list)-1):
            if list[i] == list[j]:
                print("found duplicate {0} , at position {1}".format(l[i],j))

def unique(list):
    return set(list)

find_duplicates(l)
print(list(unique(l)))

# time complexity = o(n2)